alchemy-web/docs/providers/cloudflare/openauth.md

```diff
@@ -0,0 +1,270 @@
+
+OpenAuth provides a secure, edge-compatible authentication solution with support for GitHub, Google, Discord, Facebook, and other OAuth providers. It integrates seamlessly with Cloudflare KV for session storage and Hono for API routing.
+
+## Minimal Example
+
+Create a basic OpenAuth worker with GitHub authentication:
+
+```ts
+import { OpenAuth, KVNamespace, secret } from "alchemy/cloudflare";
+
+const authStore = await KVNamespace("auth-sessions", {
+  title: "auth-sessions"
+});
+
+const auth = await OpenAuth("auth", import.meta, {
+  providers: {
+    github: {
+      clientId: secret.env.GITHUB_CLIENT_ID,
+      clientSecret: secret.env.GITHUB_CLIENT_SECRET,
+      scopes: ["user:email", "read:user"]
+    }
+  },
+  storage: authStore
+});
+```
+
+## Multiple OAuth Providers
+
+Support both GitHub and Google authentication:
+
+```ts
+import { OpenAuth, KVNamespace, secret } from "alchemy/cloudflare";
+
+const authStore = await KVNamespace("auth-sessions", {
+  title: "auth-sessions"
+});
+
+const auth = await OpenAuth("auth", import.meta, {
+  providers: {
+    github: {
+      clientId: secret.env.GITHUB_CLIENT_ID,
+      clientSecret: secret.env.GITHUB_CLIENT_SECRET,
+      scopes: ["user:email"]
+    },
+    google: {
+      clientId: secret.env.GOOGLE_CLIENT_ID,
+      clientSecret: secret.env.GOOGLE_CLIENT_SECRET,
+      scopes: ["profile", "email"]
+    }
+  },
+  storage: authStore
+});
+```
+
+## Custom Success Handler
+
+Add custom post-authentication logic:
+
+```ts
+import { OpenAuth, KVNamespace, secret } from "alchemy/cloudflare";
+
+const authStore = await KVNamespace("auth-sessions", {
+  title: "auth-sessions"
+});
+
+const auth = await OpenAuth("auth", import.meta, {
+  providers: {
+    github: {
+      clientId: secret.env.GITHUB_CLIENT_ID,
+      clientSecret: secret.env.GITHUB_CLIENT_SECRET
+    }
+  },
+  storage: authStore,
+  onSuccess: async (ctx, value) => {
+    // Custom user processing
+    console.log("User signed in:", value.user.login);
+    
+    return {
+      id: value.user.id.toString(),
+      name: value.user.name || value.user.login,
+      email: value.user.email,
+      avatar: value.user.avatar_url,
+      githubUsername: value.user.login
+    };
+  }
+});
+```
+
+## Custom API Routes
+
+Add custom routes to the Hono app:
+
+```ts
+import { OpenAuth, KVNamespace, secret } from "alchemy/cloudflare";
+
+const authStore = await KVNamespace("auth-sessions", {
+  title: "auth-sessions"
+});
+
+const auth = await OpenAuth("auth", import.meta, {
+  providers: {
+    github: {
+      clientId: secret.env.GITHUB_CLIENT_ID,
+      clientSecret: secret.env.GITHUB_CLIENT_SECRET
+    }
+  },
+  storage: authStore,
+  routes: {
+    "/api/me": `
+      return c.json({ 
+        user: c.get("user"), 
+        authenticated: true 
+      });
+    `,
+    "/api/logout": `
+      // Custom logout logic
+      return c.json({ success: true });
+    `
+  }
```

Comment:
I don't like this. Instead, `OpenAuth` should return the `app: Hono` as a property that the user is free to use.

```ts
const openAuth = OpenAuth("auth", {
  // ..
});

openAuth.app.post("/api/logout", ..)
```